app/gui/src/integration_test.rs

```diff
@@ -0,0 +1,155 @@
+
+/// A fixture for each IDE integration tests. During setup, the executor and [`Ide`] structure are
+/// initialized.
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct Fixture {
+    pub executor: EventLoopExecutor,
+    pub ide:      Ide,
+    pub root_div: HtmlDivElement,
+}
+
+impl Fixture {
+    /// Initializes the executor and `Ide` structure and returns new Fixture.
+    pub async fn new(open_to_project_view: bool) -> Self {
+        let executor = setup_global_executor();
+        let root_div = enso_web::document.create_div_or_panic();
+        root_div.set_id("root");
+        root_div.set_style_or_warn("display", "none");
+        enso_web::document.body_or_panic().append_or_warn(&root_div);
+
+        let config = crate::config::Startup { open_to_project_view, ..default() };
+        let initializer = crate::ide::Initializer::new(config);
+        let ide = initializer.start().await.expect("Failed to initialize the application.");
+        ide.ensogl_app.set_screen_size_for_tests();
+        Self { executor, ide, root_div }
+    }
+
+    /// Initializes the executor and `Ide` structure (loading the welcome screen),
+    /// and returns the new Fixture.
+    pub async fn setup() -> Self {
+        Self::new(false).await
+    }
+
+    /// Create a fixture for testing in a newly-created project.
+    ///
+    /// Initializes the IDE, creates a new project, and waits until the project is ready.
+    pub async fn setup_new_project() -> Self {
+        let fixture = Self::new(true).await;
+        fixture.new_project().await;
+        fixture
+    }
+
+    /// After returning, the IDE is in a state with new project opened and ready to work
+    /// (after libraries' compilation).
+    pub async fn new_project(&self) {
+        let project = self.ide.presenter.view().project();
+        let controller = self.ide.presenter.controller();
+        let project_management =
+            controller.manage_projects().expect("Cannot access Managing Project API");
+
+        let expect_prompt = project.show_prompt.next_event();
+        project_management.create_new_project(None).await.expect("Failed to create new project");
+        expect_prompt.await;
+    }
+
+    /// Get the Project View.
+    pub fn project_view(&self) -> crate::view::project::View {
+        self.ide.presenter.view().project()
+    }
+
+    /// Get the Graph Editor.
+    pub fn graph_editor(&self) -> crate::view::graph_editor::GraphEditor {
+        self.project_view().graph().clone_ref()
+    }
+}
+
+impl Drop for Fixture {
+    fn drop(&mut self) {
+        self.root_div.remove();
+    }
+}
+
+
+
+// ==================
+// === WaitAFrame ===
+// ==================
+
+/// A future that resolves after one frame.
+#[derive(Default, Debug)]
+#[cfg_attr(not(target_arch = "wasm32"), allow(dead_code))]
+pub struct WaitAFrame {
+    frame_passed: Rc<Cell<bool>>,
+    closure:      Option<Closure<dyn FnMut(f64)>>,
+}
+
+impl Future for WaitAFrame {
+    type Output = ();
+
```

Comment:
Just out of curiosity - i dont see any part of the codebase that use this (I was searching trough my local copy) – is this used?

Reply:
It is used to implement `pub fn wait_a_frame()`, which is used in the existing integration tests.

Reply:
Ah, its used by integration tests? I missed it. Thanks!